apis/spaces/v1alpha1/incontrolplanedefault_types.go

```diff
@@ -0,0 +1,284 @@
+}
+
+// +kubebuilder:object:root=true
+
+// InControlPlaneOverrideList is a list of InControlPlaneDefault objects.
+type InControlPlaneOverrideList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []InControlPlaneDefault `json:"items"`
+}
+
+// PatchPropagationMode denotes the traversal direction on
+// an object's hierarchy.
+type PatchPropagationMode string
+
+const (
+	// PatchPropagateAscend denotes traversal on a target object hierarchy
+	// following the metadata.ownerReferences.
+	PatchPropagateAscend PatchPropagationMode = "Ascending"
+	// PatchPropagateDescend denotes traversal on a target object hierarchy
+	// following the spec.resourceRef & spec.resourceRefs reference fields.
+	PatchPropagateDescend PatchPropagationMode = "Descending"
+	// PatchPropagateNone denotes that no traversal will be done and
+	// only the target object will be visited.
+	PatchPropagateNone PatchPropagationMode = "None"
+)
+
+// PatchDeletionPolicy controls what happens when an InControlPlaneDefault
+// object is deleted. We either attempt to roll back the changes on the
+// target object hierarchy, or we keep them.
+type PatchDeletionPolicy string
+
+const (
+	// PatchDeletionRollBack attempts to roll back the changes on
+	// the target object hierarchy when the InControlPlaneDefault object is
+	// being deleted.
+	PatchDeletionRollBack PatchDeletionPolicy = "RollBack"
+	// PatchDeletionKeep keeps the changes on the target object
+	// hierarchy when the InControlPlaneDefault object is being deleted.
+	PatchDeletionKeep PatchDeletionPolicy = "Keep"
+)
+
+// MetadataPatch represents the Kube object metadata.
+type MetadataPatch struct {
+	// Annotations represents the Kube object annotations.
+	// Only the following annotations are allowed to be patched:
+	// - crossplane.io/paused
+	// - spaces.upbound.io/force-reconcile-at
+	// +kubebuilder:validation:XValidation:rule="self.all(k, k == 'crossplane.io/paused' || k == 'spaces.upbound.io/force-reconcile-at')",message="Only the crossplane.io/paused and spaces.upbound.io/force-reconcile-at annotations are allowed"
+	// +kubebuilder:validation:MinProperties=1
+	// +kubebuilder:validation:MaxProperties=2
+	Annotations map[string]string `json:"annotations,omitempty"`
+}
+
+// Patch represents a configuration patch which is serialized into JSON to
+// obtain the fully specified intent to be used with server side apply.
+// +kubebuilder:validation:MinProperties=1
+type Patch struct {
+	// +optional
+	Metadata *MetadataPatch `json:"metadata,omitempty"`
+}
+
+// InControlPlaneOverrideSpec defines a configuration override
+// on a target object hierarchy in a target ControlPlane with the
+// given name.
+type InControlPlaneOverrideSpec struct {
+	// ControlPlaneName is the name of the target ControlPlane where
+	// the resource configuration overrides will be applied.
+	// +kubebuilder:validation:MinLength=1
+	ControlPlaneName string `json:"controlPlaneName"`
```

Comment:
To be consistent with [backup](https://github.com/upbound/spaces/blob/39b51575ca05ca4083ce7249fa44e6d40b9a2587/test/e2e/templates/backupandrestore/backup/orphan/backup.yaml#L11) ? Ideally, API conventions prefer `controlPlaneRef.name` AFAIK though.

```suggestion
	ControlPlane string `json:"controlPlaneName"`
```

Reply:
Discussed this offline. The [initial one-pager](https://github.com/upbound/arch/pull/204) first followed that convention but there was a suggestion to follow the [Kube API naming conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions) and rename the field to `controlPlaneName`. From the referenced doc:
```
The name of a field referring to another resource of kind Foo by name should be called fooName. The name of a field referring to another resource of kind Foo by ObjectReference (or subset thereof) should be called fooRef.
```

We will also open an issue to change the field name in `Backup` APIs before they become GA.